src/main/java/com/toptrumps/core/engine/GameEngine.java

```diff
@@ -16,116 +15,180 @@
+     * @return a player list containing the players who possessed
+     * a top card with the highest selected attribute value
+     */
     public List<Player> getWinners(Attribute selectedAttribute, List<Player> players) {
-        // Attribute Comparison
-        // TODO: Optimize this process -> try to loop only once and retrieve the player with max selected attribute value
+        // Retrieves the selected attribute name
         String attributeName = selectedAttribute.getName();
 
-        for (Player p : players){ // TODO Maybe integrate this for loop with something else to prevent looping.
-            Logger.getInstance().logToFileIfEnabled(p.getName() + "'s top card:  "+ p.getTopCard().toString() + "\n\n"
+        /*
+        // TODO: Move this away from the core engine.
+        for (Player p : players) { // TODO Maybe integrate this for loop with something else to prevent looping.
+            Logger.getInstance().logToFileIfEnabled(p.getName() + "'s top card:  " + p.getTopCard().toString() + "\n\n"
                     + p.getName() + "'s deck: \n" + p.toString());
         }
+        */
 
-        // Find maxValue
+        // Finds the maximum attribute value
         int maxValue = players.stream()
                 .map(p -> p.getTopCard().getAttributeByName(attributeName))
                 .max(Attribute::compareTo)
                 .get().getValue();
 
-        // Find playersWithMaxValue
-        return (ArrayList<Player>) players.stream()
+        // Returns a list of players with the maximum attribute value
+        return players.stream()
                 .filter(p -> p.getTopCard().getAttributeByName(attributeName).getValue() == maxValue)
                 .collect(toCollection(ArrayList::new));
     }
 
+    /**
+     * Returns the game outcome based on the winners and players of a round.
+     * The RoundOutcome result can be VICTORY, DRAW, or GAME_OVER.
+     *
+     * @param winners the list of players which won the round
+     * @param players the list of players present on the game
+     * @return RoundOutcome an object that encapsulates the round result, winner,
+     * number of draws, and list of players to be removed from the game
+     */
     public RoundOutcome processRoundOutcome(List<Player> winners, List<Player> players) {
-        // Round outcome
         RoundOutcome outcome;
 
-
-        //collect any defeated players and remove them from the game
+        // Collects any defeated players
         List<Player> removedPlayers = players
                 .stream()
                 .filter(p -> p.getDeckCount() == 0)
                 .collect(toCollection(ArrayList::new));
 
         if (players.size() == 1) {
+            // Checks if there is only one player left;
+            // Declares Game Over
             outcome = new RoundOutcome(GAME_OVER);
         } else if (winners.size() == 1) {
+            // Checks if there's only one winner;
+            // Sets the winner player as "active"
```

Comment:
winning? or keep as winner if you want to match with the Player object below x

Reply:
I don't get it!

Reply:
So for good english i would write "sets the winning player as active", but I realize that below this you have a Player object called winner which you are referring to in the comment.
Another solution may be "sets the Player winner as active"

Reply:
I see!!! Nice one.
(Indeed I followed the object, but it's not clear at all)